frontend/src/App.tsx

```diff
@@ -69,7 +74,17 @@ const App = (): React.ReactElement => {
         >
           <Router>
             <Switch>
-              <Route exact path={Routes.WELCOME_PAGE} component={Welcome} />
+              <Route
+                exact
+                path={Routes.WELCOME_PAGE}
+                render={() => {
+                  return authenticatedUser ? (
+                    <Redirect to={Routes.HOME_PAGE} />
+                  ) : (
+                    <Welcome />
+                  );
```

Comment:
definitely the right fix! - probably cleaner to move this check into the `Welcome` component though (similar to how the `Login` and `Signup` components redirect to the homepage)